src/runtime/graph_executor/debug/graph_executor_debug.cc

```diff
@@ -114,15 +94,69 @@ class GraphExecutorDebug : public GraphExecutor {
 
     std::ostringstream os;
     for (size_t index = 0; index < time_sec_per_op.size(); index++) {
-      os << time_sec_per_op[index] << ",";
+      double time = time_sec_per_op[index];
+      // To have good behavior when calculating total time, etc.
+      if (isnan(time)) {
+        time = 0;
+      }
+      os << time << ",";
     }
     return os.str();
   }
 
+  std::vector<std::vector<double>> RunIndividualNode(int node_index, int number, int repeat,
+                                                     int min_repeat_ms) {
+    // warmup run
+    // GraphExecutor::Run();
+    std::string tkey = module_->type_key();
+
+    // results_in_seconds[a][b] is the bth index run of the ath index repeat
+    std::vector<std::vector<double>> results_in_seconds;
+
+    if (tkey == "rpc") {
+      LOG(FATAL) << "RPC measurements should not use RunIndividualNode!";
+    }
+
+    for (int i = 0; i < repeat; ++i) {
+      std::vector<Timer> op_timers;
+      double duration_ms = 0.0;
+
+      // Keep timing operations, upping number of repeats until we reach min_repeat_ms
+      do {
+        op_timers.clear();
+        if (duration_ms > 0.0) {
+          number = static_cast<int>(std::max((min_repeat_ms / (duration_ms / number) + 1),
+                                             number * 1.618));  // 1.618 is chosen by random
+        }
+
+        std::chrono::time_point<std::chrono::high_resolution_clock, std::chrono::nanoseconds>
+            tbegin, tend;
+        tbegin = std::chrono::high_resolution_clock::now();
```

Comment:
I know you just moved the code, but this should really use the timers interface. Or better yet, just call out to time_evaluator.

Reply:
Example?

Reply:
Oh you mean get rid of all of this timing code here and instead use the python time_evaluator interface?

Reply:
Hmm, in the refactor this needs to return some timing info (it can't be outside measuring in). Do you have example of the timers interface (or just the file to look, `timers` is kind of hard to grep for).

Reply:
Here is the main loop from time evaulator: https://github.com/apache/tvm/blob/main/src/runtime/rpc/rpc_module.cc#L370-L403. Here is the timer interface: https://github.com/apache/tvm/blob/main/include/tvm/runtime/profiling.h#L43-L142.

Using time_evaluator is probably a better choice than reimplementing the timing loop.

Reply:
Done, it now uses the linked time evaluator. Lots of deduped code B)